### ISSUE TITLE
SSL exception

### ISSUE THREAD

**Issue by hjoshi123** [kiwix/kiwix-android#675]
### Are you ready to report?
   - Yes

### Environment
  * version of the software you use : 2.2
  * device / operating system : Android Nougat 7.1.2
  
### The Bug:

  * What do you obtain? A copy of the error message or a screenshot is always useful.
```
org.kiwix.kiwixmobile D/kiwixdownloadservice: Download Attempt Failed [1] times
                                                                             javax.net.ssl.SSLException: Read error: ssl=0x8dad7680: I/O error during system call, Software caused connection abort
                                                                                 at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
                                                                                 at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:758)
                                                                                 at okio.Okio$2.read(Okio.java:139)
                                                                                 at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
                                                                                 at okio.RealBufferedSource.read(RealBufferedSource.java:46)
                                                                                 at okhttp3.internal.http1.Http1Codec$AbstractSource.read(Http1Codec.java:360)
                                                                                 at okhttp3.internal.http1.Http1Codec$FixedLengthSource.read(Http1Codec.java:404)
                                                                                 at okio.RealBufferedSource.read(RealBufferedSource.java:140)
                                                                                 at okio.RealBufferedSource.read(RealBufferedSource.java:117)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$downloadChunk$9$DownloadService(DownloadService.java:451)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService$$Lambda$10.subscribe(Unknown Source)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.drain(ObservableConcatMap.java:218)
                                                                                 at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.onNext(ObservableConcatMap.java:128)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:436)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onSubscribe(ObservableFlatMap.java:546)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFromIterable.subscribeActual(ObservableFromIterable.java:55)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.tryEmit(ObservableFlatMap.java:262)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onNext(ObservableFlatMap.java:559)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:67)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$getMetaLinkContentLength$8$DownloadService(DownloadService.java:370)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService$$Lambda$9.subscribe(Unknown Source)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
                                                                                 at io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:108)
                                                                                 at io.reactivex.internal.operators.observable.ObservableRetryWhen$RepeatWhenObserver.onNext(ObservableRetryWhen.java:100)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:51)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
                                                                                 at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:43)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
03-27 18:58:46.297 26988-27009/org.kiwix.kiwixmobile D/kiwixdownloadservice:     at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
                                                                                 at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:463)
                                                                                 at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
                                                                                 at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
                                                                                 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                                 at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
                                                                                 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                                 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                                 at java.lang.Thread.run(Thread.java:761)
```
  * When does this occur? 
      Put a zim file to download and then go to airplane mode. This exception occurs without closing the app.
#### Steps to reproduce:

  1. Click on a file to download
  2. Switch to airplane mode.
  3. You get a SSL exception.
  
  * What should be the behaviour from your point of view? 
     The exception shouldn't occur.

**Comment by hjoshi123** [comment]
@kelson42 Can I work on this?

**Comment by brijeshshah13** [comment]
@hjoshi123 Are you sure that you are not receiving it because of the device because I'm getting the following exception:

```
org.kiwix.kiwixmobile D/kiwixdownloadservice: Download Attempt Failed [1] times
                                                                             okhttp3.internal.http2.StreamResetException: stream was reset: PROTOCOL_ERROR
                                                                                 at okhttp3.internal.http2.Http2Stream$FramingSource.checkNotClosed(Http2Stream.java:436)
                                                                                 at okhttp3.internal.http2.Http2Stream$FramingSource.read(Http2Stream.java:338)
                                                                                 at okhttp3.internal.http2.Http2Codec$StreamFinishingSource.read(Http2Codec.java:210)
                                                                                 at okio.RealBufferedSource.read(RealBufferedSource.java:140)
                                                                                 at okio.RealBufferedSource.read(RealBufferedSource.java:117)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$downloadChunk$9$DownloadService(DownloadService.java:451)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService$$Lambda$10.subscribe(Unknown Source)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.drain(ObservableConcatMap.java:218)
                                                                                 at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.onNext(ObservableConcatMap.java:128)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:436)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onSubscribe(ObservableFlatMap.java:546)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFromIterable.subscribeActual(ObservableFromIterable.java:55)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.tryEmit(ObservableFlatMap.java:262)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onNext(ObservableFlatMap.java:559)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:67)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$getMetaLinkContentLength$8$DownloadService(DownloadService.java:370)
                                                                                 at org.kiwix.kiwixmobile.downloader.DownloadService$$Lambda$9.subscribe(Unknown Source)
                                                                                 at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
                                                                                 at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
                                                                                 at io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:108)
                                                                                 at io.reactivex.internal.operators.observable.ObservableRetryWhen$RepeatWhenObserver.onNext(ObservableRetryWhen.java:100)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:51)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
                                                                                 at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:43)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
                                                                                 at io.reactivex.Observable.subscribe(Observable.java:11194)
                                                                                 at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
                                                                                 at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:463)
                                                                                 at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
                                                                             	at io.reactivex.i
```

**Comment by hjoshi123** [comment]
@brijeshshah13 This is the part of exception that I got. The full exception is the one I posted. Are you sure you did airplane mode?

**Comment by brijeshshah13** [comment]
@hjoshi123 Positive. Btw, I can't locate the exception I'm getting in the log you provided. I'm pretty sure they are different.

**Comment by hjoshi123** [comment]
@srv-twry Can you check if you are getting the exception?

**Comment by kelson42** [comment]
Do we have such kind of exceptions with the git master version?

**Comment by hjoshi123** [comment]
@kelson42 I am waiting for confirmation so that I can put a PR

**Comment by srv-twry** [comment]
@hjoshi123 I am AFK so i can't confirm it. If you're getting it then please send the PR.

**Comment by harshvardhan-takawale** [comment]
@hjoshi123 @brijeshshah13 I repeated the steps mentioned and was able to get the exception almost everytime. But sometimes it did throw the exception mentioned by @brijeshshah13 (like once in ten times)
```
04-09 23:42:57.658 26023-26154/org.kiwix.kiwixmobile D/kiwixdownloadservice: Download Attempt Failed [1] times
    javax.net.ssl.SSLException: Read error: ssl=0x7b66cfc0: I/O error during system call, Software caused connection abort
        at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
        at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:758)
        at okio.Okio$2.read(Okio.java:139)
        at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
        at okio.RealBufferedSource.read(RealBufferedSource.java:46)
        at okhttp3.internal.http1.Http1Codec$AbstractSource.read(Http1Codec.java:360)
        at okhttp3.internal.http1.Http1Codec$FixedLengthSource.read(Http1Codec.java:404)
        at okio.RealBufferedSource.read(RealBufferedSource.java:140)
        at okio.RealBufferedSource.read(RealBufferedSource.java:117)
        at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$downloadChunk$9(DownloadService.java:451)
        at org.kiwix.kiwixmobile.downloader.-$$Lambda$DownloadService$TfEBu5_LbV4R9Y8gYGOPSHOsdaw.subscribe(lambda)
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.drain(ObservableConcatMap.java:218)
        at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.onNext(ObservableConcatMap.java:128)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:436)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onSubscribe(ObservableFlatMap.java:546)
        at io.reactivex.internal.operators.observable.ObservableFromIterable.subscribeActual(ObservableFromIterable.java:55)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.tryEmit(ObservableFlatMap.java:262)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onNext(ObservableFlatMap.java:559)
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:67)
        at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$getMetaLinkContentLength$8(DownloadService.java:370)
        at org.kiwix.kiwixmobile.downloader.-$$Lambda$DownloadService$FcfqijxUwoyhso-xshLx8NoH4yc.subscribe(lambda)
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
        at io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:108)
        at io.reactivex.internal.operators.observable.ObservableRetryWhen$RepeatWhenObserver.onNext(ObservableRetryWhen.java:100)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:51)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:43)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:463)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
```


```
10 00:16:04.395 3579-4640/org.kiwix.kiwixmobile D/kiwixdownloadservice: Download Attempt Failed [1] times
    okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL
        at okhttp3.internal.http2.Http2Stream$FramingSource.checkNotClosed(Http2Stream.java:436)
        at okhttp3.internal.http2.Http2Stream$FramingSource.read(Http2Stream.java:338)
        at okhttp3.internal.http2.Http2Codec$StreamFinishingSource.read(Http2Codec.java:210)
        at okio.RealBufferedSource.read(RealBufferedSource.java:140)
        at okio.RealBufferedSource.read(RealBufferedSource.java:117)
        at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$downloadChunk$9(DownloadService.java:451)
        at org.kiwix.kiwixmobile.downloader.-$$Lambda$DownloadService$TfEBu5_LbV4R9Y8gYGOPSHOsdaw.subscribe(lambda)
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.drain(ObservableConcatMap.java:218)
        at io.reactivex.internal.operators.observable.ObservableConcatMap$SourceObserver.onNext(ObservableConcatMap.java:128)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drainLoop(ObservableFlatMap.java:436)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.drain(ObservableFlatMap.java:323)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onSubscribe(ObservableFlatMap.java:546)
        at io.reactivex.internal.operators.observable.ObservableFromIterable.subscribeActual(ObservableFromIterable.java:55)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.tryEmit(ObservableFlatMap.java:262)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$InnerObserver.onNext(ObservableFlatMap.java:559)
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:67)
        at org.kiwix.kiwixmobile.downloader.DownloadService.lambda$getMetaLinkContentLength$8(DownloadService.java:370)
        at org.kiwix.kiwixmobile.downloader.-$$Lambda$DownloadService$FcfqijxUwoyhso-xshLx8NoH4yc.subscribe(lambda)
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.subscribeInner(ObservableFlatMap.java:162)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:139)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
        at io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:108)
        at io.reactivex.internal.operators.observable.ObservableRetryWhen$RepeatWhenObserver.onNext(ObservableRetryWhen.java:100)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:51)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:43)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:11194)
        at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:463)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
```

**Comment by brijeshshah13** [comment]
@hjoshi123 Please update your local master and try again to reproduce the exception.

**Comment by hjoshi123** [comment]
@brijeshshah13 I updated it and its still showing me the exception.

**Comment by kelson42** [comment]
@macgills @mhutti1 I strongly suspect we could close this after 2.5 implementation (fixes) and more particuliary #510?

**Comment by mhutti1** [comment]
I think so
